data/core/start.lua

```diff
@@ -19,3 +19,18 @@ package.path = DATADIR .. '/?.lua;' .. package.path
 package.path = DATADIR .. '/?/init.lua;' .. package.path
 package.path = USERDIR .. '/?.lua;' .. package.path
 package.path = USERDIR .. '/?/init.lua;' .. package.path
+
+local dynamic_suffix = MACOS and 'lib' or (WINDOWS and 'dll' or 'so')
+package.cpath = DATADIR .. '/?.' .. dynamic_suffix .. ';' .. package.cpath
```

Comment:
the initial `.. package.cpath` should be dopped here, I think using the system default should be avoided. This is also true for `package.path` but that is not as big of a problem.

Reply:
Re: This and the next comment. So, in theory, if you were to require a native plugin that didn't interface with lua, really; it would still be valid. Like if requiring a plugin just ran a program. Or, if you require a plugin with the exact same system lib as your statically linked library, I think it would probably work; which may make easier for people using luarocks to have system modules.

I dunno if that's a compelling reason to *not* do this though. I do see the reason why we should disallow that. Does anyone have any other thoughts? I think that @Mehgugs is probably correct, though. It probably should be disallowed, if no one has any objections.

Reply:
It's still possible to link to a shared object by path from lua using `package.loadlib` if people are so determined, this would just prevent the automatic behaviour